src/systems/a320_systems/src/pneumatic.rs

```diff
@@ -1,27 +1,3041 @@
+
+        self.exhaust
+            .update_move_fluid(context, &mut self.pack_container);
+    }
+}
+impl PneumaticContainer for PackComplex {
+    fn pressure(&self) -> Pressure {
+        self.pack_container.pressure()
+    }
+
+    fn volume(&self) -> Volume {
+        self.pack_container.volume()
+    }
+
+    fn temperature(&self) -> ThermodynamicTemperature {
+        self.pack_container.temperature()
+    }
+
+    fn change_fluid_amount(&mut self, volume: Volume) {
+        self.pack_container.change_fluid_amount(volume);
+    }
+
+    fn update_temperature(&mut self, temperature_change: TemperatureInterval) {
+        self.pack_container.update_temperature(temperature_change);
+    }
+}
+impl SimulationElement for PackComplex {
+    fn accept<T: SimulationElementVisitor>(&mut self, visitor: &mut T) {
+        self.pack_flow_valve_controller.accept(visitor);
+
+        visitor.visit(self);
+    }
+
+    fn write(&self, writer: &mut SimulatorWriter) {
+        writer.write(
+            &self.pack_flow_valve_flow_rate_id,
+            self.pack_flow_valve.fluid_flow(),
+        );
+    }
+}
+
+// In the future, this will be done by the ACSC, hence why I have used an external controller and not the BMC
+struct PackFlowValveController {
+    pack_toggle_pb_id: VariableIdentifier,
+    pack_pb_is_auto: bool,
+    pid: PidController,
+}
+impl PackFlowValveController {
+    fn new(context: &mut InitContext, engine_number: usize) -> Self {
+        Self {
+            pack_toggle_pb_id: context
+                .get_identifier(format!("AIRCOND_PACK{}_TOGGLE", engine_number)),
+            pack_pb_is_auto: true,
+            pid: PidController::new(0., 0.05, 0., 0., 1., 0.75),
+        }
+    }
+
+    fn update(&mut self, context: &UpdateContext, pack_flow_valve_flow_rate: VolumeRate) {
+        self.pid.next_control_output(
+            pack_flow_valve_flow_rate.get::<cubic_meter_per_second>(),
+            Some(context.delta()),
+        );
+    }
+}
+impl ControllerSignal<PackFlowValveSignal> for PackFlowValveController {
+    fn signal(&self) -> Option<PackFlowValveSignal> {
+        Some(if self.pack_pb_is_auto {
+            PackFlowValveSignal::new(Ratio::new::<ratio>(self.pid.output()))
+        } else {
+            PackFlowValveSignal::new_closed()
+        })
+    }
+}
+impl SimulationElement for PackFlowValveController {
+    fn read(&mut self, reader: &mut SimulatorReader) {
+        self.pack_pb_is_auto = reader.read(&self.pack_toggle_pb_id);
+    }
+}
+
+/// This is a unique valve (and specific to the A320 probably) because it is controlled by two motors. One for manual control and one for automatic control
+pub struct CrossBleedValve {
+    open_amount: Ratio,
+    connector: PneumaticContainerConnector,
+    spring_characteristic: f64,
+    is_powered_for_manual_control: bool,
+    is_powered_for_automatic_control: bool,
+}
+impl CrossBleedValve {
+    pub fn new(spring_characteristic: f64) -> Self {
```

Comment:
**question**: Are you expecting differences in `spring_characteristic`? Currently the only usage initialises this with `1.`